woocommerce/woocommerce-ios
# Edit Coupon: fields formatting, validation and errors management

Part of #6488 

### Description
Here I implemented a way in the `TitleAndTextFieldRow` reusable component to accept an optional `UnitInputFormatter` for formatting the text following the rules of the input formatted.
- I implemented also some custom input formatters and I used also the existing ones for formatting the text in the fields of the Edit Coupon screen and in the Usage Restrictions screen.
- In the end, I implemented a local coupon validator, that will display a notice in case of error.

### Testing instructions
1. Navigate to a Coupon detail, and start editing it.
2. Try to edit the Coupon Amount -> You should be able to type only the price, and use just the separator (`,` or `.`) based on the settings of your store.
3. Try to type a coupon code -> You should be able to type just uppercased strings.
4. Try to remove completely the coupon code and press "Save" -> You should see a Notice error.
5. Open the usage restrictions screen.
6. In the minimum spending field, you should be able to type only the price, and use just the separator (`,` or `.`) based on the settings of your store.
7. In the maximum spending field, you should be able to type only the price, and use just the separator (`,` or `.`) based on the settings of your store.
8. In the other text fields, you should be able to delete the current content and type just integer numbers.


---
